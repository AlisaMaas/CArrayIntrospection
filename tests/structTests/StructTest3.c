/**
* Try doing some accesses to two 
* separate structs, some of which set
* struct fields based on other struct 
* fields.
**/

struct foo {
	int a;
	double b;
	struct foo *c;
};

struct bar {
	struct foo first;
	int x;
	double y;
	struct bar *z;
};

void accessBoth(struct foo f, struct bar b) {
	b.x = f.a;
	b.y = f.a * f.b;
	b.z = &b;
	b.first = f;
}