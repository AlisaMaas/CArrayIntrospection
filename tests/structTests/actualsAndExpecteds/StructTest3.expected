struct.foo:
	element 0 accessed
	element 1 accessed
struct.bar:
	element 0 accessed
	element 1 accessed
	element 2 accessed
	element 3 accessed
